hazelcast/src/test/java/com/hazelcast/spi/impl/operationexecutor/classic/ClassicOperationExecutorTest.java

```diff
@@ -83,4 +85,28 @@ public void run() throws Exception {
             }
         });
     }
+
+    @Test(expected = NullPointerException.class)
+    public void test_runOnAllPartitionThreads_whenTaskNull() {
+        initExecutor();
+        executor.runOnAllPartitionThreads(null);
+    }
+
+    @Test
+    public void test_runOnAllPartitionThreads() {
+        initExecutor();
+
+        int threadCount = executor.getPartitionOperationThreadCount();
+        final CountDownLatch latch = new CountDownLatch(threadCount);
+
+        executor.runOnAllPartitionThreads(new Runnable() {
+            @Override
+            public void run() {
+                latch.countDown();
+            }
+        });
+
+        assertOpenEventually(latch);
```

Comment:
That's indeed quite clever. Now please pass me some eye bleach please! :)

I would say a `CyclicBarrier` would make your intent clearer. 
Let's create a cyclic barrier with `threadCount + 1`. and use `await` instead of `assertOpenEventually`. ?